src/main/java/uk/gov/hmcts/ccd/data/casedetails/supplementarydata/SupplementaryDataQueryBuilder.java

```diff
@@ -0,0 +1,45 @@
+package uk.gov.hmcts.ccd.data.casedetails.supplementarydata;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.vladmihalcea.hibernate.type.json.JsonNodeBinaryType;
+import java.time.LocalDateTime;
+import java.time.ZoneOffset;
+import java.util.List;
+import java.util.Map;
+import javax.persistence.EntityManager;
+import javax.persistence.Query;
+import org.hibernate.query.NativeQuery;
+
+public interface SupplementaryDataQueryBuilder {
+
+    ObjectMapper mapper = new ObjectMapper();
+
+    SupplementaryDataProcessor dataProcessor = new SupplementaryDataProcessor();
+
+    List<Query> buildQueries(EntityManager entityManager, String caseReference, Map<String, Object> requestData);
+
+    Operation operationType();
+
+    default String requestJson(Object supplementaryData) {
```

Comment:
I think this could be a `.toJson()` method on the `SupplementaryDataRequest`

Reply:
renamed